c2rust-refactor/src/scripting.rs

```diff
@@ -601,5 +641,144 @@ impl<'a, 'tcx> UserData for TransformCtxt<'a, 'tcx> {
+            });
+
+            Ok(LuaAstNode::new(expr))
+        });
+
+        methods.add_method("ident_path_ty", |_lua_ctx, _this, path: LuaString| {
+            let path = syntax::ast::Path::from_ident(Ident::from_str(path.to_str()?));
+            let expr = P(Ty {
+                id: DUMMY_NODE_ID,
+                node: TyKind::Path(None, path),
+                span: DUMMY_SP,
+            });
+
+            Ok(LuaAstNode::new(expr))
+        });
+
+        methods.add_method("vec_mac_init_num", |_lua_ctx, _this, (init, num): (LuaAstNode<P<Expr>>, LuaAstNode<P<Expr>>)| {
+            let init = Rc::new(Nonterminal::NtExpr(init.borrow().clone()));
+            let num = Rc::new(Nonterminal::NtExpr(num.borrow().clone()));
+            let macro_body = vec![
+                TokenTree::Token(Token { kind: TokenKind::Interpolated(init), span: DUMMY_SP }),
+                TokenTree::Token(Token { kind: TokenKind::Semi, span: DUMMY_SP }),
+                TokenTree::Token(Token { kind: TokenKind::Interpolated(num), span: DUMMY_SP}),
+            ];
+            let mac = mk().mac(mk().path("vec"), macro_body, MacDelimiter::Bracket);
+            let expr = P(Expr {
+                id: DUMMY_NODE_ID,
+                node: ExprKind::Mac(mac),
+                span: DUMMY_SP,
+                attrs: ThinVec::new(),
+            });
+
+            Ok(LuaAstNode::new(expr))
+        });
+
+        methods.add_method("int_lit_expr", |_lua_ctx, _this, (int, _suffix): (i64, Option<LuaString>)| {
+            let lit = Lit {
+                token: TokenLit {
+                    kind: TokenLitKind::Integer,
+                    symbol: Symbol::intern(&format!("{}", int)),
+                    suffix: None,
+                },
+                node: LitKind::Int(int as u128, LitIntType::Unsuffixed),
+                span: DUMMY_SP,
+            };
+            let expr = P(Expr {
+                id: DUMMY_NODE_ID,
+                node: ExprKind::Lit(lit),
+                span: DUMMY_SP,
+                attrs: ThinVec::new(),
+            });
+
+            Ok(LuaAstNode::new(expr))
+        });
+
+        methods.add_method("cast_expr", |_lua_ctx, _this, (expr, ty): (LuaAstNode<P<Expr>>, LuaAstNode<P<Ty>>)| {
+            let expr = expr.borrow().clone();
+            let ty = ty.borrow().clone();
+            let expr = P(Expr {
+                id: DUMMY_NODE_ID,
+                node: ExprKind::Cast(expr, ty),
+                span: DUMMY_SP,
+                attrs: ThinVec::new(),
+            });
+
+            Ok(LuaAstNode::new(expr))
+        });
+
+        methods.add_method("get_expr_ty", |_lua_ctx, this, expr: LuaAstNode<P<Expr>>| {
+            let rty = this.cx.node_type(expr.borrow().id);
+            let ty = reflect_tcx_ty(this.cx.ty_ctxt(), rty);
+
+            Ok(LuaAstNode::new(ty))
+        });
+
+        methods.add_method("get_field_expr_hirid", |_lua_ctx, this, expr: LuaAstNode<P<Expr>>| {
```

Comment:
Not sure - the field id, in this case, is a `DefId`, not a `NodeId`. I've managed to avoid exposing `DefId`s to Lua until now. Do you think we should expose it?